helpers/inference.rb

```diff
@@ -1,17 +1,30 @@
 # frozen_string_literal: true
 
 class Clover
+  def visible_capable_models(dataset)
+    dataset
+      .where(visible: true)
+      .where(Sequel.pg_jsonb_op(:tags).get_text("capability") => ["Text Generation", "Embeddings"])
+      .order(:model_name)
+  end
+
   def inference_endpoint_ds
     dataset_private = dataset_authorize(@project.inference_endpoints_dataset, "InferenceEndpoint:view")
     dataset_public = InferenceEndpoint.where(is_public: true)
 
     dataset = dataset_private.union(dataset_public)
-    dataset = dataset.where(visible: true)
-    dataset = dataset.where(Sequel.pg_jsonb_op(:tags).get_text("capability") => ["Text Generation", "Embeddings"])
-    dataset = dataset.order(:model_name)
+    dataset = visible_capable_models(dataset)
     dataset.eager(:load_balancer)
   end
 
+  def inference_router_model_ds
+    visible_capable_models(InferenceRouterModel)
+      .select_all(:inference_router_model)
+      .join(:inference_router_target, inference_router_model_id: :id)
+      .eager(inference_router_targets: {inference_router: :load_balancer})
+      .distinct
```

Comment:
```suggestion
      .eager_graph(inference_router_targets: {inference_router: :load_balancer})
      .exclude(inference_router_model_id: nil)
```

`eager_graph` uses a JOIN approach (LEFT JOIN by default, so you need to exclude `NULL` values).  You could change which column you are checking for NULL from `:inference_router_model_id` to `Sequel[:load_balancer][:id]` if you only want router models that have a related load balancer.

Reply:
I got undefined method 'inference_router' for nil error with eager_graph. I think we want show model with a target. A target should always have a load balancer. The current code works. What is the benefit of using eager_graph?

Reply:
It's odd to use join/distinct to check the associated object exists, and then also use separate queries for eager loading.  With eager_graph, there are no separate queries, just a single query that uses JOINs.  You can use a Database logger (DATABASE_LOGGER_LEVEL=info in the environment) if you want to see the SQL queries generated.

Not sure why you would be getting a NoMethodError with a change from `eager` to `eager_graph`, any chance you saved the backtrace?

Reply:
Done. I think everything works now.